mscore/pianoview.cpp

```diff
@@ -1242,4 +1296,133 @@ void PianoView::setSubdiv(int value)
+                        firstTick = startTick;
+                        init = true;
+                        }
+                  }
+            }
+
+      //No valid notes
+      if (!init)
+            return;
+
+      QString xmlStrn;
+      QXmlStreamWriter xml(&xmlStrn);
+//      xml.setAutoFormatting(true);
+      xml.writeStartDocument();
+
+      xml.writeStartElement("notes");
+      xml.writeAttribute("firstTickN", QString::number(firstTick.numerator()));
+      xml.writeAttribute("firstTickD", QString::number(firstTick.denominator()));
+
+      //bundle notes into XML file & send to clipboard.
+      //This is only affects pianoview and is not part of the regular copy/paste process
+      for (int i = 0; i < noteList.size(); ++i) {
+            if (noteList[i]->note()->selected()) {
+                  Note* note = noteList[i]->note();
+
+                  Chord* chord = note->chord();
+
+                  Fraction ticks = chord->ticks();
+                  Fraction tieLen = Fraction(note->playTicks(), 1) - ticks;
+                  Fraction len = ticks + tieLen;
+
+                  Fraction startTick = note->chord()->tick();
+                  int pitch = note->pitch();
+
+                  int voice = note->voice();
+
+                  xml.writeStartElement("note");
+                  xml.writeAttribute("startTickN", QString::number(startTick.numerator()));
+                  xml.writeAttribute("startTickD", QString::number(startTick.denominator()));
+                  xml.writeAttribute("tickLenN", QString::number(len.numerator()));
+                  xml.writeAttribute("tickLenD", QString::number(len.denominator()));
+                  xml.writeAttribute("pitch", QString::number(pitch));
+                  xml.writeAttribute("voice", QString::number(voice));
+                  xml.writeEndElement();
+                  }
+            }
+
+      xml.writeEndElement();
+      xml.writeEndDocument();
+
+      QMimeData* mimeData = new QMimeData;
+      mimeData->setData(PIANO_NOTE_MIME_TYPE, xmlStrn.toUtf8());
+      QApplication::clipboard()->setMimeData(mimeData);
+      }
+
+//---------------------------------------------------------
+//   pasteNotesAtCursor
+//---------------------------------------------------------
+
+void PianoView::pasteNotesAtCursor()
+      {
+      //ScoreView::normalPaste();
+      const QMimeData* ms = QApplication::clipboard()->mimeData();
+      if (!ms)
+            return;
+
+      int pickTick = pixelXToTick(popupMenuPos.x());
+      int pickPitch = pixelYToPitch(popupMenuPos.y());
+//      int pickTick = pixelXToTick(mouseDownPos.x());
+//      int pickPitch = pixelYToPitch(mouseDownPos.y());
+      int subbeats = _tuplet * (1 << _subdiv);
+      int subbeatTicks = MScore::division / subbeats;
+      int roundedTick = (pickTick / subbeatTicks) * subbeatTicks;
+
+      Score* score = _staff->score();
+      score->startCmd();
+
+      if (ms->hasFormat(PIANO_NOTE_MIME_TYPE)) {
```

Comment:
Does this static string appear elsewhere, like in the preference keys in `global` module? If so, better not make a duplicate here (difficult in keeping in sync)

Reply:
it is used here (and later 2 more times), not set. That is done further up in this file. Not used anywhere else.

Reply:
Then OK